src/os/ui/timelinepanel.js

```diff
@@ -52,12 +52,20 @@ os.ui.resizeMap = 'resizeMap';
 os.ui.TimelinePanelCtrl = function($scope, $element, $timeout) {
   os.ui.TimelinePanelCtrl.base(this, 'constructor', $scope, $element, $timeout);
 
+  /**
+   * @type {boolean}
+   */
+  this['locked'] = os.time.TimelineController.getInstance().getLock();
+
+  this.tlc.listen(os.time.TimelineEventType.LOCK_TOGGLE, goog.partial(this.setLock, true), false, this);
```

Comment:
This `listen` call needs a corresponding `unlisten` in this directive's destroy function.

Also, `this.setLock` is being bound with an argument of `true`, but that function doesn't take any arguments.